normandy/recipes/api/serializers.py

```diff
@@ -102,6 +104,25 @@ class Meta:
             'is_approved',
         ]
 
+    def validate_arguments(self, value):
+        # Get the schema associated with the selected action
+        schema = Action.objects.get(name=self.initial_data.get('action')).arguments_schema
+        schemaValidator = Validator(schema)
+
+        errorResponse = []
+        errors = sorted(schemaValidator.iter_errors(value), key=lambda e: e.path)
+
+        for error in errors:
+            errorResponse.append({
+                'message': error.message,
+                'field': json.JSONEncoder().encode(list(error.path))
+            })
+
+        if len(errorResponse):
+            raise serializers.ValidationError(errorResponse)
+
+        return value
+
```

Comment:
Is this where this should go? I tried putting the validation logic all in `/validators.py` but I wasn't able to import the `Action` model there for some reason, which I needed in order to query the schema.